core/src/main/java/org/apache/iceberg/EnvironmentContext.java

```diff
@@ -26,7 +26,9 @@ public class EnvironmentContext {
   public static final String ENGINE_NAME = "engine-name";
   public static final String ENGINE_VERSION = "engine-version";
 
-  private EnvironmentContext() {}
+  private EnvironmentContext() {
+    PROPERTIES.put("iceberg-version", IcebergBuild.fullVersion());
```

Comment:
Should we move this to a constant like the other properties above?

Reply:
I thought about that, but it's not really used anywhere, other than this place